csi.proto

```diff
@@ -1020,6 +1023,54 @@ message NodeUnpublishVolumeRequest {
 message NodeUnpublishVolumeResponse {
   // Intentionally empty.
 }
+message NodeGetVolumeStatsRequest {
+  // The ID of the volume. This field is REQUIRED.
+  string volume_id = 1;
+
+  // The path at which the volume was published. It MUST be an absolute
+  // path in the root filesystem of the process serving this request.
+  // This is a REQUIRED field.
+  string target_path = 2;
+}
+
+message NodeGetVolumeStatsResponse {
+  // The used capacity in bytes. This field is OPTIONAL.
+  // The value of this field MUST NOT be negative.
+  // For volumes that share a filesystem with the host (e.g hostpath)
+  // this is used space on the underlying storage by specified
+  // shared path or directory.
+  int64 used_capacity = 1;
+
+  // The available capacity in bytes. This field is OPTIONAL.
+  // For volumes that share a filesystem with the host (e.g hostpath)
+  // this is available space on the underlying storage and is
+  // shared with the host processes and other volumes.
+  // The value of this field MUST NOT be negative.
+  int64 available_capacity = 2;
+
+  // The total capacity in bytes. This field is REQUIRED.
+  // For volumes that share a file system with the host (e.g hostpath)
+  // this is the total size of underlying storage and it MAY
+  // not be equal to used_capacity + available_capacity because
+  // filesystem is shared.
+  // The value of this field MUST NOT be negative.
+  int64 total_capacity = 3;
+
+  // The number of inodes used by the volume.
+  // This field is OPTIONAL.
+  // The value of this field MUST NOT be negative.
+  int64 used_inodes = 4;
```

Comment:
I agree with @ddebroy that we have to avoid leaking OS specific constructs in to the spec. Is there a similar concept in Windows, could we come up with something that would work for all OSes?

Reply:
Yeah I think our options boil down to:

a. Either come up with a name that may or may not represent inode number in a OS agnostic way. It seems hard to do and could easily cause confusion.
b. Drop the inode information altogether from stat information.
d. Keep the inode capacity information as it is and make it clear that plugin authors need not set this value when running on a platform that does not support it. These numbers will not be available when using raw block devices on *nix too. So, Windows is not alone in this.
e. @msau42 suggested we can move such platform specific information in a opaque dictionary. It could work but I am not sure if it is very different from option#d.